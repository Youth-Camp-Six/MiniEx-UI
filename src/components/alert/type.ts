export type Alert = {
    type: string;
    title: string;
    content: string;
};
export interface AlertProps {
    list?: Alert[];
    className?: string
}