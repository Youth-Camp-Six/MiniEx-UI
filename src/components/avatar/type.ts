import { ReactNode } from 'react';
type status = 'online' | 'processing' | 'away' | 'offline' | 'busy' | '';

export interface AvatarProps {
  className?: string;
  round?: string;
  status?: status;
  src?: string;
  alt?: string;
  width?: string;
  height?: string;
  statusPosition?: string;
  icon?: ReactNode;
  style?: React.CSSProperties;
  children?: ReactNode;
}
type baseOptions = {
  label: string;
  value: string;
};
export type Options = AvatarProps & baseOptions;
export interface AvatarGroupProps {
  className?: string;
  maxCount?: number;
  maxPopoverPlacement?: 'top' | 'bottom';
  maxPopoverTrigger?: 'hover' | 'focus' | 'click';
  maxStyle?: React.CSSProperties;
  options: Options[];
  width?: string;
  height?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}
