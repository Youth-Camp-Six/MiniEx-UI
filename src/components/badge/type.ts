import { ReactNode } from 'react';

type value = string | number | 'undefined';

type type = 'normal' | 'success' | 'error' | 'warning' | 'info';

export interface BadgeProps {
    className?: string;
    round?: string;
    value?: value;
    type?: type;
    children: ReactNode;
}
