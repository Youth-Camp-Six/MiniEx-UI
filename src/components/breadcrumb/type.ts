export type Options = {
    label: string;
    to: string;
    active?: boolean;
};

type iconType = 'sprit' | 'shoulder'
export interface breadcrumbProps {
    className?: string;
    options: Options[];
    iconType?: iconType;
}