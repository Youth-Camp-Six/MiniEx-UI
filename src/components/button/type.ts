import { ReactNode } from 'react';

type ButtonSize = 'lg' | 'sm';

type ButtonType = 'primary' | 'default' | 'danger' | 'link';

export interface BaseButtonProps {
  className?: string;
  disabled?: boolean;
  size?: ButtonSize;
  btnType?: ButtonType;
  children: ReactNode;
}

// HTML button元素的属性结合上自定义属性
type buttonProps = BaseButtonProps & React.ButtonHTMLAttributes<HTMLElement>;

// 将Button组件的所有属性转化为可选的
export type ButtonProps = Partial<buttonProps>;
