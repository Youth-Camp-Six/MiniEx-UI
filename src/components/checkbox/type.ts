import { ReactNode } from 'react';
export interface CheckboxProps {
  onChange?: (checked: boolean) => void;
  label: string;
  block?: boolean;
  iconable?: boolean;
  checked?: boolean;
  round?: boolean;
  size?: string;
  children?: ReactNode;
  className?: string;
  disabled?: boolean;
}

type Options = {
  label: string;
  value: string | number;
};
export interface CheckboxGroupProps {
  onChange?: (value: any) => void;
  value?: Array<string | number>;
  block?: boolean;
  iconable?: boolean;
  options: Options[];
  direction?: 'x' | 'y';
  size?: string;
  round?: boolean;
  children?: ReactNode;
  className?: string;
}
