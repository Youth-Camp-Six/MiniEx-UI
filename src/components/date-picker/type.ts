import { Dayjs } from 'dayjs';

// type Region = 'CN' | 'US';
// interface WeekKeys extends Record<Region, string> {
//   CN: '一' | '二' | '三' | '四' | '五' | '六' | '日';
//   US: 'Mo' | 'Tu' | 'We' | 'Th' | 'Fr' | 'Sa' | 'Su';
// }
export type TimeDimension = 'year' | 'month' | 'date';

export interface IDatePickerContext {
  weekFirstDay: 1 | 2 | 3 | 4 | 5 | 6 | 7;
  displayTime: Dayjs;
  value: Dayjs;
  dimension: TimeDimension;
}

export interface IDatePickerProps {
  // for context
  weekFirstDay?: 1 | 2 | 3 | 4 | 5 | 6 | 7;
  defaultValue?: Dayjs;
  type?: TimeDimension;
  // others
  validator?: (date: Dayjs) => boolean;
}

export interface IDatePickerHeader {
  onDimensionChange: (type: TimeDimension) => void;
  onChange: (value: Dayjs) => void;
}
export interface IDateCellProps {
  value: Dayjs;
  onClick: (value: Dayjs) => void;
  isThisMonth: boolean;
  disable?: boolean;
  selected?: boolean;
}

export interface IDateCellsProps {
  onClick: (value: Dayjs) => void;
  validator?: (date: Dayjs) => boolean;
}
