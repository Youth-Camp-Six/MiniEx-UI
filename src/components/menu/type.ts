import { ReactNode } from 'react';
type MenuMode = 'horizontal' | 'vertical';

export interface MenuProps {
  defaultIndex?: number;
  className?: string;
  mode?: MenuMode;
  style?: React.CSSProperties;
  onSelect?: (selectedIndex: number) => void;
  children?: ReactNode;
}

export interface MenuItemProps {
  index?: number;
  disabled?: boolean;
  className?: string;
  style?: React.CSSProperties;
  children?: ReactNode;
}

export interface IMenuContext {
  index: number;
  onSelect?: (selectedIndex: number) => void;
}
