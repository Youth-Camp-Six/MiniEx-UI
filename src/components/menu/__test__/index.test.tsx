/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, fireEvent, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Menu, MenuItem } from '..';
import { MenuProps } from '../type';

const menuProps: MenuProps = {
  onSelect: jest.fn(),
  defaultIndex: 0,
  className: 'test',
};

const generateMenu = (props) => (
  <Menu {...props}>
    <MenuItem index={0}>actived item0</MenuItem>
    <MenuItem index={1} disabled>
      disabled item1
    </MenuItem>
    <MenuItem index={2}>item2</MenuItem>
  </Menu>
);

let wrapper: RenderResult,
  menuElement: HTMLElement,
  activeElement: HTMLElement,
  disabledElement: HTMLElement;

describe('test menu comonent', () => {
  beforeEach(() => {
    wrapper = render(generateMenu(menuProps));
    menuElement = wrapper.getByTestId('test-menu');
    activeElement = wrapper.getByText('actived item0');
    disabledElement = wrapper.getByText('disabled item1');
  });

  it('menu snapshot test', () => {
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it('default props', () => {
    expect(menuElement).toBeInTheDocument();
    expect(menuElement).toHaveClass('menu test');
    expect(menuElement.getElementsByTagName('li').length).toEqual(3);
    expect(activeElement).toHaveClass('menu-item is-active');
    expect(disabledElement).toHaveClass('menu-item is-disabled');
  });

  it('active and right callback', () => {
    const thirdItem = wrapper.getByText('item2');
    fireEvent.click(thirdItem);
    expect(thirdItem).toHaveClass('is-active');
    expect(activeElement).not.toHaveClass('is-active');
    expect(menuProps.onSelect).toHaveBeenCalledWith(2);
    fireEvent.click(disabledElement);
    expect(disabledElement).not.toHaveClass('is-active');
    expect(menuProps.onSelect).not.toHaveBeenCalledWith(1);
  });
});
