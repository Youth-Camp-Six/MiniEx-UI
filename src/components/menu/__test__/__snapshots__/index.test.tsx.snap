// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button comonent button snapshot test 1`] = `
<ul
  class="menu test"
  data-testid="test-menu"
>
  <li
    class="menu-item is-active"
  >
    actived item0
  </li>
  <li
    class="menu-item is-disabled"
  >
    disabled item1
  </li>
  <li
    class="menu-item"
  >
    item2
  </li>
</ul>
`;

exports[`test button comonent menu snapshot test 1`] = `
<ul
  class="menu test"
  data-testid="test-menu"
>
  <li
    class="menu-item is-active"
  >
    actived item0
  </li>
  <li
    class="menu-item is-disabled"
  >
    disabled item1
  </li>
  <li
    class="menu-item"
  >
    item2
  </li>
</ul>
`;

exports[`test menu comonent menu snapshot test 1`] = `
<ul
  class="menu test"
  data-testid="test-menu"
>
  <li
    class="menu-item is-active"
  >
    actived item0
  </li>
  <li
    class="menu-item is-disabled"
  >
    disabled item1
  </li>
  <li
    class="menu-item"
  >
    item2
  </li>
</ul>
`;
